physicsnemo/utils/sdf.py

```diff
@@ -112,41 +135,32 @@ def signed_distance_field(
     >>> signed_distance_field(mesh_vertices, mesh_indices, input_points)
     array([0.5], dtype=float32)
     """
-    wp.init()
-
-    # Get the current warp device string (e.g., "cuda:2")
-    try:
-        device = str(wp.get_device())
-    except Exception:
-        device = "cuda"
+    if return_cupy is None:
+        return_cupy = any(
+            isinstance(arr, cp.ndarray)
+            for arr in (mesh_vertices, mesh_indices, input_points)
+        )
 
-    # If cupy arrays come in, we have to convert them to numpy arrays:
-    return_cupy = False
-    if isinstance(mesh_vertices, cp.ndarray):
-        return_cupy = True
-    if isinstance(mesh_indices, cp.ndarray):
-        return_cupy = True
-    if isinstance(input_points, cp.ndarray):
-        return_cupy = True
-
-    # Convert numpy to warp arrays:
-    mesh_vertices = wp.array(mesh_vertices, dtype=wp.vec3, device=device)
-    mesh_indices = wp.array(mesh_indices, dtype=wp.int32, device=device)
+    wp.init()
+    device = wp.get_device()
 
-    sdf_points = wp.array(input_points, dtype=wp.vec3, device=device)
+    mesh = wp.Mesh(
+        points=wp.array(mesh_vertices, dtype=wp.vec3f, device=device),
+        indices=wp.array(mesh_indices, dtype=wp.int32, device=device),
+    )
 
-    mesh = wp.Mesh(mesh_vertices, mesh_indices)
+    N = len(input_points)
 
-    sdf = wp.zeros(shape=sdf_points.shape, dtype=wp.float32, device=device)
-    sdf_hit_point = wp.zeros(shape=sdf_points.shape, dtype=wp.vec3f, device=device)
-    sdf_hit_point_id = wp.zeros(shape=sdf_points.shape, dtype=wp.int32, device=device)
+    sdf = wp.empty(shape=(N,), dtype=wp.float32, device=device)
+    sdf_hit_point = wp.empty(shape=(N,), dtype=wp.vec3f, device=device)
+    sdf_hit_point_id = wp.empty(shape=(N,), dtype=wp.int32, device=device)
```

Comment:
Does the sdf kernel below hit every point here?  or is it possible some of these points won't be visited by warp?

If its possible that not every single output location will be assigned in the kernel, we should init to the default values and not empty tensors, which is probably undefined in values.